godot-core/src/builtin/file.rs

```diff
@@ -0,0 +1,459 @@
+        self.fa.store_double(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_float(&mut self, value: f32) -> IoResult<()> {
+        self.fa.store_float(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_real(&mut self, value: f32) -> IoResult<()> {
+        self.fa.store_real(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_string(&mut self, value: GodotString) -> IoResult<()> {
+        self.fa.store_string(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_pascal_string(&mut self, value: GodotString) -> IoResult<()> {
+        self.fa.store_pascal_string(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_line(&mut self, value: GodotString) -> IoResult<()> {
+        self.fa.store_line(value);
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_csv_line(
+        &mut self,
+        values: PackedStringArray,
+        delim: GodotString,
+    ) -> IoResult<()> {
+        self.fa.store_csv_line_ex(values).delim(delim).done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    pub fn store_variant(&mut self, value: Variant, full_objects: bool) -> IoResult<()> {
+        self.fa
+            .store_var_ex(value)
+            .full_objects(full_objects)
+            .done();
+        self.check_error()?;
+        Ok(())
+    }
+
+    // Varia
+    pub fn set_big_endian(&mut self, value: bool) {
+        self.fa.set_big_endian(value);
+    }
+
+    pub fn is_big_endian(&self) -> bool {
+        self.fa.is_big_endian()
+    }
+
+    pub fn get_path(&self) -> GodotString {
+        self.fa.get_path()
+    }
+
+    pub fn get_path_absolute(&self) -> GodotString {
+        self.fa.get_path_absolute()
+    }
+
+    /// Returns the current file position
+    pub fn get_position(&self) -> u64 {
+        self.fa.get_position()
+    }
+}
+
+impl Read for GFile {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        let length = self.fa.get_length();
+        let position = self.fa.get_position();
+
+        if position >= length {
+            return Ok(0);
+        }
+
+        let remaining_bytes = (length - position) as usize;
+        let bytes_to_read = cmp::min(buf.len(), remaining_bytes);
+
+        if bytes_to_read == 0 {
+            return Ok(0);
+        }
+        let mut readen = 0;
+        while readen < bytes_to_read {
+            buf[readen] = self.fa.get_8();
+            readen += 1;
+            self.check_error()?;
+        }
+        Ok(readen)
+    }
+}
+
+impl Write for GFile {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let bytes_to_write = buf.len();
+        let mut bytes_written = 0;
+
+        while bytes_written < bytes_to_write {
+            self.fa.store_8(buf[bytes_written]);
+            bytes_written += 1;
+            self.check_error()?;
+        }
```

Comment:
Note: writing individual bytes is very inefficient, I think in the future we should use this method: https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-store-buffer

We could keep a `PackedByteArray` allocated with our buffer size. Maybe we then don't even need a `Vec`. Same for reading.

Anyway, we should do this in a separate PR; I think it's great to get this initial version working first before optimizing 🙂

Reply:
If there would be a move to `store_buffer` for `write`, the `flush()` will potentially also need its implementation. But as you said - its a topic for the future :)